viz-graph doesn't work when nodes are maps

This code:
```clojure
(defn -main []
  (-> (u/digraph)
      (u/add-edges [{:a 1} {:b 2}]
                   [{:b 2} {:c 3}]
                   [{:a 1} {:c 3}])
      (u/viz-graph)))
```
fails with:
```
Exception in thread "main" java.lang.RuntimeException: Expected AST node of type #{:dorothy.core/node-id :dorothy.core/subgraph}, compiling:(/private/var/folders/w1/4drznygn7qg6fm_46bh67v2m0000gp/T/form-init174042668194109870.clj:1:124)
...
Caused by: java.lang.RuntimeException: Expected AST node of type #{:dorothy.core/node-id :dorothy.core/subgraph}
	at dorothy.core$error.doInvoke(core.clj:201)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at dorothy.core$check_ast.invoke(core.clj:264)
...
	at clojure.core$str.doInvoke(core.clj:528)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invoke(core.clj:630)
	at dorothy.core$dot_STAR__statements.invoke(core.clj:490)
	at dorothy.core$eval2979$fn__2981.invoke(core.clj:536)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at dorothy.core$dot.invoke(core.clj:557)
```
Looks like the nodes are being converted to strings here https://github.com/Engelberg/ubergraph/blob/master/src/ubergraph/core.clj#L780, but the same is not being done for the edges.
Changing the code to also convert the nodes when generating `directed-edges` and `undirected-edges` fixes this error, but fails to generate a valid graph because the node id seems to get truncated at the first colon, so any keywords in the map cause it to be corrupted.

--- a/src/ubergraph/core.clj
+++ b/src/ubergraph/core.clj
@@ -1,12 +1,12 @@
 (ns ubergraph.core
   (:require [potemkin :refer [import-vars def-map-type deftype+]]
             [potemkin.collections :refer [AbstractMap]]
-            [loom.graph :as lg] 
+            [loom.graph :as lg]
             [loom.attr :as la]
             [ubergraph.protocols :as up]
             [dorothy.core :as d]
-            [clojure.pprint]
-            ))
+            [clojure.string :as str]
+            [clojure.pprint]))
 
 (import-vars 
   [loom.graph
@@ -754,6 +754,13 @@ Undirected edges are counted only once."
           g))
       $ (edges g)))) 
 
+(defn- dotid [n]
+  (if (or (string? n)
+          (keyword? n)
+          (number? n))
+    n
+    (str/replace (print-str n) ":" "")))
+
 (defn viz-graph
   "Uses graphviz to generate a visualization of your graph. Graphviz 
 must be installed on your computer and in your path. Passes along
@@ -773,17 +780,13 @@ Takes an optional map which can contain:
           ns (nodes g),
           es (edges g)
           nodes (for [n ns]
-                  [(if (or (string? n)
-                           (keyword? n)
-                           (number? n))
-                     n
-                     (print-str n))
+                  [(dotid n)
                    (sanitize-attrs g n)]),
           directed-edges (for [e es :when (directed-edge? e)]
-                           [(src e) (dest e) (sanitize-attrs g e)])
+                           [(dotid (src e)) (dotid (dest e)) (sanitize-attrs g e)])
           undirected-edges (for [e es :when (and (undirected-edge? e)
                                                  (not (mirror-edge? e)))]
-                             [(src e) (dest e)
+                             [(dotid (src e)) (dotid (dest e))
                               (merge {:dir :none} (sanitize-attrs g e))])]
       (-> (concat [{:layout layout}] 
                   nodes directed-edges undirected-edges)

